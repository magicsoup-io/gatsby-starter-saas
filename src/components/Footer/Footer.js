import React from 'react';
import { StaticQuery, graphql } from 'gatsby'
import { 
  Box,
  Flex,
  Container,
  Heading,
} from '@magicsoup.io/stock'
import { 
  P,
  Footer as StyledFooter,
} from '../../styled'

const Footer = () => (
  <StaticQuery
    query={graphql`
      query {
        generalJson {
          footer {
            elements {
              headline
              text
            }
          }
        }
      }
    `}
    render={data => 
      <StyledFooter>
        <Container pt={5} pb={4} px={3}>
          <Flex flexWrap='wrap' flexDirection='column'>
            {data.generalJson.footer.elements.map((element, index) => (
              <Box 
                key={`footer-${index}`}
                px={3}
                pt={4}
                pb={3}
                width={[1]}>
                <Heading as='h4' variant='h4' color='white'>{element.headline}</Heading>
                <P color='white' dangerouslySetInnerHTML={{__html: element.text }} />
              </Box>
            ))}
          </Flex>
        </Container>
        <Box 
          bg='primaryDark'
          >
          <Container py={3}>
            <P 
              fontSize={1}
              color='white'
              mt={0}
              textAlign='center'
              >Â© 2018 - {new Date().getFullYear()} <a href="https://www.zauberware.com" target="_blank">zauberware technologies</a></P>
          </Container>
        </Box>
      </StyledFooter>
    }
  />
)

export default Footer