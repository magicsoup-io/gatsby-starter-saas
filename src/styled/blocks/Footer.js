// styled-components based extension
import styled from 'styled-components'
import { Box } from '@magicsoup.io/stock'

const Footer = styled(Box)`
  a{
    color: white;
    text-decoration: underline;
  }
`

Footer.defaultProps = {
  as: 'footer',
  bg: 'primary',
}

export default Footer