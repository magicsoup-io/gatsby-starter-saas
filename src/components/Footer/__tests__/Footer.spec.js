import React from 'react'
import renderer from 'react-test-renderer'
import { StaticQuery } from 'gatsby'
import Footer from '../Footer'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      generalJson: {
        footer: {
          elements: [
            {
              headline: 'Home',
              text: 'A text in the footer',
            },
            {
              headline: 'Contact',
              text: 'A text in the footer',
            }
          ]
        }
      },
    })
  )
})

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})