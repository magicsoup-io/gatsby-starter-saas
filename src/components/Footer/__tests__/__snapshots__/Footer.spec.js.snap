// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="sc-bdVaJa sc-iwsKbI dJKMkL"
>
  <div
    className="sc-bdVaJa sc-bxivhb fhvOTI"
  >
    <div
      className="sc-bdVaJa sc-ifAKCX kqdpKR"
    >
      <div
        className="sc-bdVaJa hPaROM"
        width={
          Array [
            1,
          ]
        }
      >
        <h4
          className="sc-bdVaJa sc-EHOje sc-bZQynM cuCRPK"
          color="white"
        >
          Home
        </h4>
        <p
          className="sc-bdVaJa sc-EHOje sc-gqjmRU sFrRH"
          color="white"
          dangerouslySetInnerHTML={
            Object {
              "__html": "A text in the footer",
            }
          }
          fontSize={2}
        />
      </div>
      <div
        className="sc-bdVaJa hPaROM"
        width={
          Array [
            1,
          ]
        }
      >
        <h4
          className="sc-bdVaJa sc-EHOje sc-bZQynM cuCRPK"
          color="white"
        >
          Contact
        </h4>
        <p
          className="sc-bdVaJa sc-EHOje sc-gqjmRU sFrRH"
          color="white"
          dangerouslySetInnerHTML={
            Object {
              "__html": "A text in the footer",
            }
          }
          fontSize={2}
        />
      </div>
    </div>
  </div>
  <div
    className="sc-bdVaJa gzJBxY"
  >
    <div
      className="sc-bdVaJa sc-bxivhb gbgRUr"
    >
      <p
        className="sc-bdVaJa sc-EHOje sc-gqjmRU ivXjAx"
        color="white"
        fontSize={1}
      >
        Â© 2018 - 
        2019
         
        <a
          href="https://www.zauberware.com"
          target="_blank"
        >
          zauberware technologies
        </a>
      </p>
    </div>
  </div>
</footer>
`;
